packages/apps/desktop-viewer-test/src/backend/main.ts

```diff
@@ -60,10 +63,81 @@ const viewerMain = async () => {
   if (process.env.NODE_ENV === "development") {
     ElectronHost.mainWindow?.webContents.toggleDevTools();
   }
+  // add the menu
+  ElectronHost.mainWindow?.on("ready-to-show", createMenu);
+};
+
+const createMenu = () => {
+  const isMac = process.platform === "darwin";
+
+  const template = [
+    {
+      label: "File",
+      submenu: [
+        {
+          label: "Open Snapshot File",
+          click: () => {
+            IpcHost.send(channelName, "snapshot");
+          },
+        },
+        {
+          label: "View Remote iModel",
+          click: () => {
+            IpcHost.send(channelName, "remote");
+          },
+        },
+        { type: "separator" },
+        isMac ? { role: "close" } : { role: "quit" },
+      ],
+    },
+    {
+      label: "View",
+      submenu: [
+        {
+          label: "Getting Started",
+          click: () => {
+            IpcHost.send(channelName, "home");
+          },
+        },
+      ],
+    },
+    {
+      label: "Window",
+      submenu: [
+        {
+          label: "Minimize",
+          role: "minimize",
+        },
+        {
+          label: "Zoom",
+          role: "zoom",
+        },
+      ],
+    },
+  ] as MenuItemConstructorOptions[];
+
+  if (isMac) {
+    template.unshift({
+      label: "iTwin Viewer",
+      role: "appMenu",
+      submenu: [
+        {
+          label: "Preferences",
+          click: () => {
+            IpcHost.send(channelName, "preferences");
+          },
+        },
+      ],
+    } as MenuItemConstructorOptions);
+  }
+
+  const menu = Menu.buildFromTemplate(template as MenuItemConstructorOptions[]);
+
+  Menu.setApplicationMenu(menu);
 };
 
 try {
-  viewerMain(); // eslint-disable-line @typescript-eslint/no-floating-promises
+  void viewerMain();
```

Comment:
I had the same comment on a PR that Keith recently made... I'm not a huge fan of this fix that ESLint recommends for this, it is an easy thing to miss when reviewing code and/or PRs of a `void` vs. and `await`. Plus not very obvious on what it actually does...

Reply:
I think this is the recommended approach to say this function is truly asynchronous and we do not need to wait on it. This is not a case where we can or want to use await or top level await. The only alternatives that I am aware of are empty catches or disabling the rule. I feel like void is more meaningful and cleaner than both of those? https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md